src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1132,10 +1133,62 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
     private boolean promptUser(PromptBehavior prompt) {
         return prompt == PromptBehavior.Always || prompt == PromptBehavior.REFRESH_SESSION;
     }
+    
+    /**
+     * check the redirectUri before sending the request
+     * if the redirectUri from the client does not match the valid redirectUri
+     * the client app would not jump to the login page
+     * redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE
+     * 
+     * @param callbackHandle
+     * @param request
+     * @throws UnsupportedEncodingException
+     * @return true if the RedictUri is valid
+     */
+    private boolean verifyBrokerRedirectUri(CallbackHandler callbackHandle, final AuthenticationRequest request){
+        final String methodName = ":verifyBrokerRedirectUri";
+        String errMsg = new String();        
+        String inputUri = request.getRedirectUri();
+        PackageHelper helper = new PackageHelper(mContext);
+        
+        if(StringExtensions.IsNullOrBlank(inputUri))
+        {
+            errMsg = "the redirectUri is null or blank";            
+        }
+        else if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://"))
+        {
+            errMsg = "the redirectUri is not starting with the valid redict prefix " + AuthenticationConstants.Broker.REDIRECT_PREFIX;            
+        } else
+			try {
+				if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://" + URLEncoder.encode(mContext.getPackageName(), AuthenticationConstants.ENCODING_UTF8) + "/"))
+				{
+				    errMsg = "the package name of redirectUri is not as expected. The expected package name is" + URLEncoder.encode(mContext.getPackageName(), AuthenticationConstants.ENCODING_UTF8);
+				}
+				else if(!inputUri.equalsIgnoreCase(getRedirectUriForBroker()))
+				{
+				    errMsg = "the signature of redirectUri is not as expected. The expected signiture is " + URLEncoder.encode(helper.getCurrentSignatureForPackage(mContext.getPackageName()), AuthenticationConstants.ENCODING_UTF8);
+				}
+			} catch (UnsupportedEncodingException e) {
+				Logger.e(TAG + methodName, "Digest error", "", ADALError.ENCODING_IS_NOT_SUPPORTED, e);
+				callbackHandle.onError(new AuthenticationException(ADALError.ENCODING_IS_NOT_SUPPORTED, "Digest error"));
+			}
+        if(!errMsg.isEmpty())
```

Comment:
Add a space before this line to make it clear that it is not part of the previous set of clauses.